arch/xtensa/core/xtensa-asm2.c

```diff
@@ -147,6 +154,7 @@ static inline unsigned int get_bits(int offset, int num_bits, unsigned int val)
 __unused void *xtensa_int##l##_c(void *interrupted_stack)	\
 {							   \
 	uint32_t irqs, intenable, m;			   \
+	usage_stop();					   \
```

Comment:
why are we doing this for xtensa only? what about other architectures?

Reply:
It's an optimization.  It could absolutely be done for any architecture, it just hasn't been done in this PR.  Again, the current implementation doesn't do this at all.

Reply:
> Again, the current implementation doesn't do this at all.

right, and the purpose of this PR is to add this optimization, and it needs to be done for all architectures.

Reply:
Again, I don't understand the depth of the criticism here.  This PR does lots of good stuff not related to ISR fidelity in measurements.  And we have optional optimizations everywhere in the architecture API (atomics, switch-vs-swap, sched IPIs, stuff like that).  And this optimization is limited to USE_SWITCH architectures anyway, we can't do it everywhere because there's no portable ISR exit hook on the legacy scheme.

I'll see about adding the architectures I know well, but demanding that new features not be merged except by someone willing to hook the ISR entry code of every architecture in the tree doesn't seem like it's going to scale.